test/labeling/model/test_baseline_model.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+
+import numpy as np
+
+from snorkel.labeling.model.baselines import (
+    MajorityClassVoter,
+    MajorityLabelVoter,
+    RandomVoter,
+)
+
+
+class BaselineModelTest(unittest.TestCase):
+    def test_random_vote(self):
+        L = np.array([[1, 1, 3, 1], [2, 4, 1, 2], [1, 3, 1, 2]]).T
+        rand_voter = RandomVoter()
+        rand_voter.train_model()
+        Y_p = rand_voter.predict_proba(L)
+        self.assertLessEqual(Y_p.max(), 1.0)
+        self.assertGreaterEqual(Y_p.min(), 0.0)
+
+    def test_majority_class_vote(self):
+        L = np.array([[1, 1, 2, 1], [2, 2, 1, 2], [1, 1, 1, 2]]).T
+        mc_voter = MajorityClassVoter()
+        mc_voter.train_model(balance=[0.8, 0.2])
+        Y_p = mc_voter.predict_proba(L)
+
+        Y_p_true = np.array([[1.0, 0.0], [1.0, 0.0], [1.0, 0.0], [1.0, 0.0]])
+        self.assertTrue((Y_p == Y_p_true).all())
+
+    def test_majority_label_vote(self):
+        L = np.array([[1, 1, 2, 1], [2, 2, 1, 2], [1, 1, 1, 2]]).T
+        ml_voter = MajorityLabelVoter()
+        ml_voter.train_model(balance=[0.8, 0.2])
```

Comment:
Unused

Reply:
A little out-of-scope but baselines.py is a mess. Should be

```python
import numpy as np

from .label_model import LabelModel

class BaselineVoter(LabelModel):
    def train_model(self, *args, **kwargs):
        pass

class RandomVoter(BaselineVoter):
    """
    A class that votes randomly among the available labels
    """

    def predict_proba(self, L):
        """
        Args:
            L: An [n, m] scipy.sparse matrix of labels
        Returns:
            output: A [n, k] np.ndarray of probabilistic labels
        """
        n = L.shape[0]
        Y_p = np.random.rand(n, self.k)
        Y_p /= Y_p.sum(axis=1).reshape(-1, 1)
        return Y_p


class MajorityClassVoter(LabelModel):
    """
    A class that places all probability on the majority class based on class
    balance (and ignoring the label matrix).

    Note that in the case of ties, non-integer probabilities are possible.
    """

    def train_model(self, balance, *args, **kwargs):
        """
        Args:
            balance: A 1d arraylike that sums to 1, corresponding to the
                (possibly estimated) class balance.
        """
        self.balance = np.array(balance)

    def predict_proba(self, L):
        n = L.shape[0]
        Y_p = np.zeros((n, self.k))
        max_classes = np.where(self.balance == max(self.balance))
        for c in max_classes:
            Y_p[:, c] = 1.0
        Y_p /= Y_p.sum(axis=1).reshape(-1, 1)
        return Y_p


class MajorityLabelVoter(BaselineVoter):
    """
    A class that places all probability on the majority label from all
    non-abstaining LFs for that task.

    Note that in the case of ties, non-integer probabilities are possible.
    """

    def predict_proba(self, L):
        L = self._to_numpy(L).astype(int)
        n, m = L.shape
        Y_p = np.zeros((n, self.k))
        for i in range(n):
            counts = np.zeros(self.k)
            for j in range(m):
                if L[i, j]:
                    counts[L[i, j] - 1] += 1
            Y_p[i, :] = np.where(counts == max(counts), 1, 0)
        Y_p /= Y_p.sum(axis=1).reshape(-1, 1)
        return Y_p
```

Reply:
MajorityClassVoter and MajorityLabelVoter are not "RandomVoters". And `train_model` inheritance made no sense.